tests/cancun/eip4844_blobs/test_blob_txs.py

```diff
@@ -396,27 +426,86 @@ def expected_excess_blob_gas(
 
 
 @pytest.fixture
-def blocks(
-    expected_blob_gas_used: Optional[int | Removable],
+def header_verify(
+    txs: List[Transaction],
+    expected_blob_gas_used_from_transition_tool: Optional[int | Removable],
     expected_excess_blob_gas: Optional[int | Removable],
+) -> Header:
+    """
+    Header fields to verify from the transition tool.
+    """
+    header_verify = Header()
+    header_verify.blob_gas_used = expected_blob_gas_used_from_transition_tool
+    header_verify.excess_blob_gas = expected_excess_blob_gas
+    if len([tx for tx in txs if not tx.error]) == 0:
+        header_verify.gas_used = 0
+    return header_verify
+
+
+@pytest.fixture
+def all_blob_gas_used(
+    fork: Fork,
+    txs: List[Transaction],
+    block_number: int,
+    block_timestamp: int,
+) -> Optional[int | Removable]:
+    """
+    Calculates the blob gas used by the test block taking into account failed transactions.
+    """
+    if not fork.header_blob_gas_used_required(
+        block_number=block_number, timestamp=block_timestamp
+    ):
+        return Header.EMPTY_FIELD
+    return sum([Spec.get_total_blob_gas(tx) for tx in txs])
+
+
+@pytest.fixture
+def rlp_modifier(
+    all_blob_gas_used: Optional[int | Removable],
+) -> Optional[Header]:
+    """
+    Header fields to modify on the output block in the BlockchainTest.
+    """
+    if all_blob_gas_used == Header.EMPTY_FIELD:
+        return None
+    return Header(
+        blob_gas_used=all_blob_gas_used,
+    )
+
+
+@pytest.fixture
+def test_block(
     txs: List[Transaction],
     block_error: Optional[str],
     engine_api_error_code: Optional[EngineAPIError],
+    header_verify: Optional[Header],
+    rlp_modifier: Optional[Header],
+) -> Block:
+    """
+    Test block for all test cases.
+    """
+    return Block(
+        txs=txs,
+        exception=block_error,
+        engine_api_error_code=engine_api_error_code,
+        header_verify=header_verify,
+        rlp_modifier=rlp_modifier,
+    )
+
+
+@pytest.fixture
+def blocks(
```

Comment:
We could remove `blocks` here only retaining `test_block`, assuming the latter below